### ISSUE TITLE
Flux and Helm Operator - refresh, update

### ISSUE THREAD

**Issue by kingdonb** [okteto/charts#109]
This should really be two issues but I don't want to spam your repo...

Flux (v1) and Helm Operator have both seen releases since this chart was published (okteto/charts/flux)

They could most likely easily be refreshed, (and if you assigned this issue to me I will follow it up and contribute the changes myself 👍) - unless you get there first. There wouldn't be any breaking changes in Flux v1, so it should be very straightforward to update it to the latest stable release.

There is a bigger update needed for Flux v2, which is a total rewrite from the ground up and a completely new design. This would require additional support from Okteto Cloud and/or Enterprise to provide a migration path for users to the next version. (By this I mean, Okteto Enterprise now installs and Cloud bundles the HelmRelease CRD from Helm Operator, it has been updated to `helm.toolkit.fluxcd.io/v2beta1` for now, with a number of others required for full Flux operation, and beta releases for all that are still in alpha state, coming soon.) Now that Flux v2 is at 100% feature parity, we should be able to recommend to upgrade to all users (although some will inevitably wait for the "2.0" stable release, we hope many will migrate earlier so we can solve any new or surprising issues that might require a breaking changes, before it is declared too late.)

Flux v1 is still in maintenance mode, but users are actively encouraged to migrate to Flux v2 with the [GitOps Toolkit](https://toolkit.fluxcd.io) and it would be extremely beneficial for Flux to have Okteto's support in helping users migrate. This will not be entirely straightforward at first, as your flux v1 onboarding flow is extremely well orchestrated and nailed down. It would be a separate issue to provide an onboarding flow for Okteto with as great an experience and all the same features enabled by default and right out of the box, like for instance setting up webhooks. 💯

**Comment by rberrelleza** [comment]
@kingdonb this is great! if flux v2 is now API compatible with flux v1, do you think that we could just replace v1 with v2 in the catalog?

**Comment by kingdonb** [comment]
API compatible is a stretch. It is all new APIs, the feature parity is from an end-user perspective.

Flux v2 is a breaking update in a lot of ways.

Helm controller is pretty close to API compatible, but Flux never had an API (not for all the things it does now) or CRDs other than HelmRelease.

**Comment by kingdonb** [comment]
The change in #110 is definitely API compatible, as Flux v1 doesn't really have an API (other than for fluxctl, which has a very limited API surface.)

I am trying to find a user who can repro fluxcd/flux#3199 as it has been reported as a regression in the latest versions of Flux v1, ... you can push out this upgrade now, or hold it until the next release. Another release of Flux v1 is imminent with Kubernetes client version upgrades (and hopefully one more soon after that, with fixes for any outstanding issues affecting current Flux v1 users).

**Comment by kingdonb** [comment]
Glad to see you've taken the next step and updated Helm Operator as well

**Comment by rberrelleza** [comment]
need to do final testing, but it looks like it's working fine @kingdonb. Once #114 is done let's close this issue, and we can file a separate one to discuss flux v2, I think there's a lot of think about for that one (including https://github.com/okteto/okteto/issues/1253)

**Comment by kingdonb** [comment]
I think that sounds good. I haven't found any show-stoppers in Flux 1.21.2 and the next minor release is due out soon.

I can add it to my release checklist to come back and notify Okteto with a PR when there is a new Flux v1 version out.

We are having some discussions on the Flux team about how to support Flux (v2) as a Service. I'll come back to you on Slack so we can talk about this more. I think there's more that we can do to make sure that Okteto is able to provide the same level of support or greater for Flux v2, with the constraints of the new multi-tenancy model.

I am not sure how to resolve this yet, but I'm sure we can come up with something that makes sense.

**Comment by kingdonb** [comment]
1.22.0 release is due out today.

If this change is still pending release in okteto/charts, then I can still update it today after the release is cut.

**Comment by kingdonb** [comment]
No worries, now you don't have to do this twice (at least not until we do another release)

https://github.com/fluxcd/flux/pull/3449

A chart-1.7.0 tag which publishes a new chart release, should be tagged and pushed within a few minutes after that merges.

**Comment by kingdonb** [comment]
The chart 1.7.0 is published

fluxcd/flux#3449 is merged

**Comment by kingdonb** [comment]
Closing #109 as it's now duplicate of these two:

* #114
* okteto/okteto#1253

Thanks for collaborating on this release
